controllers/policyset/policyset_controller.go

```diff
@@ -203,11 +204,17 @@ func (r *PolicySetReconciler) processPolicySet(ctx context.Context, plcSet *poli
 			// aggregate compliance state
 			plcComplianceState := complianceInRelevantClusters(childPlc.Status.Status, clusters)
 			if plcComplianceState == "" {
-				pendingPolicies = append(pendingPolicies, string(childPlcName))
+				unknownPolicies = append(unknownPolicies, string(childPlcName))
 			} else {
-				compliancesFound = append(compliancesFound, string(childPlcName))
-				if plcComplianceState == "NonCompliant" {
-					aggregatedCompliance = "NonCompliant"
+				if plcComplianceState == policyv1.Pending &&
+					aggregatedCompliance != policyv1.NonCompliant {
+					pendingPolicies = append(pendingPolicies, string(childPlcName))
+					aggregatedCompliance = policyv1.Pending
+				} else {
+					compliancesFound = append(compliancesFound, string(childPlcName))
+					if plcComplianceState == policyv1.NonCompliant {
+						aggregatedCompliance = policyv1.NonCompliant
+					}
```

Comment:
I think you could simplify this by having this outside of the if/else and change it to:
```go
if aggregatedCompliance != policyv1.NonCompliant && aggregatedCompliance != policyv1.Pending {
aggregatedCompliance = plcComplianceState
}
```